src/uct/tcp/tcp_iface.c

```diff
@@ -213,15 +213,27 @@ static int uct_tcp_iface_is_reachable(const uct_iface_h tl_iface,
     return 1;
 }
 
-static const char *
-uct_tcp_iface_get_sysfs_path(const char *dev_name, char *path_buffer)
+static const char *uct_tcp_iface_get_sysfs_path(const char *dev_name,
+                                                char *path_buffer,
+                                                size_t path_buffer_size)
 {
-    char dev_path[PATH_MAX];
+    char dev_path[path_buffer_size];
 
-    ucs_snprintf_safe(dev_path, PATH_MAX, "%s/%s", UCT_TCP_IFACE_NETDEV_DIR,
-                      dev_name);
+    ucs_snprintf_safe(dev_path, sizeof(dev_path), "%s/%s",
+                      UCT_TCP_IFACE_NETDEV_DIR, dev_name);
     return uct_iface_get_sysfs_path(dev_path, dev_name, path_buffer);
 }
+
+double uct_tcp_iface_get_pci_bw(const char *dev_name)
```

Comment:
can we move uct_tcp_iface_get_sysfs_path+uct_tcp_iface_get_pci_bw+uct_iface_get_sysfs_path to ucs/topo?

Reply:
I think `uct_tcp_iface_get_sysfs_path` better stay in tcp_iface otherwise topo will have too many identical interfaces, and it will be confusing.